nix-darwin/PrepareDarwin.hs

```diff
@@ -76,10 +85,16 @@ setupSSLCert = unlessM (testfile cert) $ mapM_ sudo setup
 
 -- | Prepare /etc for use with nix-darwin instead of nix.
 prepareConfigs :: Shell ()
-prepareConfigs = do
+prepareConfigs =
   -- prepare configs for nix darwin
-  mapM_ moveAway ["/etc/bashrc", "/etc/nix/nix.conf"]
-  chopProfile "/etc/profile"
+  need "USER" >>= \case
+    Just user -> do
+      mapM_ moveAway ["/etc/nix/nix.conf"]
+      let contents = fromString $ "trusted-users = " <> T.unpack user
```

Comment:
if you get rid of both the `fromString` and the `T.unpack` it should remain as a `Text` and not need any conversion